crates/binjs_generate_library/src/lib.rs

```diff
@@ -133,15 +133,34 @@ use json::JsonValue as JSON;
 
         // Buffer used to generate the generic data structure (impl declaration).
         let mut impl_buffer = String::new();
-        impl_buffer.push_str("impl Library {\n    pub fn new(builder: &mut SpecBuilder) -> Self {\n        let names = Library {\n");
+        impl_buffer.push_str(
+            "impl Library {
+    /// Create a new Spec from this library.
+    pub fn spec() -> Spec {
+        let mut builder = SpecBuilder::new();
+        let names = Self::new(&mut builder);
+        builder.into_spec(SpecOptions {
+            null: &names.meta_null_interface,
+            root: &names.meta_entry_point,
+        })
+    }
+}
+",
+        );
+        impl_buffer.push_str(
+            "impl Library {
```

Comment:
given `spec` and `new` are defined unconditionally, you can use single `impl Library` block for them.